modules/packages/LinearAlgebra.chpl

```diff
@@ -2500,6 +2517,410 @@ proc cosm(A: []) throws {
   }
 }
 
+// Gets sums along specified axes. Method for Rectangular matrix.
+proc absSum(A: [?D], axis=0) throws where !isCSDom(D) {
+  if (axis != 0 && axis != 1) {
+    throw new LinearAlgebraError("absSum(): axis=0 or axis=1 expected");
+  }
+
+  if axis == 0 then
+    // axis=0 sums along column
+    return forall j in D.dim(1-axis) do (+ reduce abs(A[D.dim(axis), j]));
+  else
+    // axis=1 sums along row
+    return forall i in D.dim(1-axis) do (+ reduce abs(A[i, D.dim(axis)]));
+}
+
+// Gets sums along specified axes. Method for CSR matrix.
+private proc absSum(A: [?ADom], axis=0) throws where isCSDom(ADom) {
+  if (axis != 0 && axis != 1) {
+    throw new LinearAlgebraError("absSum(): axis=0 or axis=1 expected");
+  }
+
+  var (n,t) = ADom.shape;
+  if axis == 0 {
+    // axis=0 sums along column
+    var b: [0..<t] A.eltType;
+    forall (i,j) in ADom with (+ reduce b) {
+      b[j] += abs(A[i,j]);
+    }
+    return b;
+  }
+  else {
+    // axis=1 sums along rows
+    var b: [0..<n] A.eltType;
+    forall (i,j) in ADom with (+ reduce b) {
+      b[i] += abs(A[i,j]);
+    }
+    return b;
+  }
+}
+
+// Gets max along specified axes. Method for CSR matrix.
+private proc maxAlongAxis(A: [?ADom], axis=0) throws where isCSDom(ADom) {
+  if (axis != 0 && axis != 1) {
+    throw new LinearAlgebraError("absSum(): axis=0 or axis=1 expected");
+  }
+
+  var (n,t) = ADom.shape;
+  if axis == 0 {
+    // axis=0 sums along column
+    var b: [0..<t] A.eltType;
+    forall (i,j) in ADom with (max reduce b) {
+      b[j] = max(b[j], A[i,j]);
+    }
+    return b;
+  }
+  else {
+    // axis=1 sums along rows
+    var b: [0..<n] A.eltType;
+    forall (i,j) in ADom with (max reduce b) {
+      b[i] = max(b[i], A[i,j]);
+    }
+    return b;
+  }
+}
+
+// Gets max along specified axes. Method for Rectangular matrix.
+private proc maxAlongAxis(A: [?ADom], axis=0) throws where !isCSDom(ADom) {
+  if (axis != 0 && axis != 1) {
+    throw new LinearAlgebraError("absSum(): axis=0 or axis=1 expected");
+  }
+
+  var (n,t) = ADom.shape;
+  if axis == 0 {
+    // axis=0 sums along column
+    var b: [0..<t] A.eltType;
+    forall i in 0..<t {
+      b[i] = max reduce A[.., i];
+    }
+    return b;
+  }
+  else {
+    // axis=1 sums along rows
+    var b: [0..<n] A.eltType;
+    forall i in 0..<n {
+      b[i] = max reduce A[i, ..];
+    }
+    return b;
+  }
+}
+
+// Given the Domain and element-type, method
+// returns a Matrix of ones.
+private proc ones(Dom: domain(2), type eltType=real) {
+  var A: [Dom] eltType = 1 : eltType;
+  return A;
+}
+
+
+// Given the Domain and element-type, method
+// returns a Matrix of ones.
+private proc zeros(Dom: domain(2), type eltType=real) {
+  var A: [Dom] eltType = 0 : eltType;
+  return A;
+}
+
+
+// Returns an index's unit vector of size n.
+private proc elementaryVector(n,ind) {
+  var v: [0..<n] real;
+  v[ind] = 1;
+  return v;
+}
+
+record Cmp {
+  param desc;
+  var A;
+}
+
+proc Cmp.compare(a, b) {
```

Comment:
given that this record is really tiny, I prefer this method to be a primary method (defined inside the record defintion without the prefix `Cmp.`)